src/NServiceBus.AcceptanceTesting/Support/EndpointBehaviorBuilder.cs

```diff
@@ -24,26 +24,26 @@ public EndpointBehaviorBuilder<TContext> Given(Action<IBus> action)
         }
 
 
-        public EndpointBehaviorBuilder<TContext> Given(Action<IBus,TContext> action)
+        public EndpointBehaviorBuilder<TContext> Given(Func<IBus, TContext, Task> action)
         {
             behavior.Givens.Add(new GivenDefinition<TContext>(action));
 
             return this;
         }
 
-        public EndpointBehaviorBuilder<TContext> When(Action<IBus> action)
+        public EndpointBehaviorBuilder<TContext> When(Func<IBus, Task> action)
         {
             return When(c => true, action);
         }
```

Comment:
there is a single test using this When signature which could be changed to use Given. So could we remove the When without condition?